### DISCUSSION TITLE
Me touchpad middle-click paste text

### DISCUSSION THREAD

**Discussion by raffaelemancuso** [hyprwm/Hyprland#1355] [Q&A]
Me touchpad middle-click pastes the text of the clipboard.

I want to disable this as I often click it by mistake while trying to left or right click.

How can I do it?

**Comment by vaxerski** [comment]
should be some libinput opt no? @fufexan

**Comment by fufexan** [comment]
`input:touch:middle_button_emulation = 0`.

It's in the wiki, in Variables, Input section.

EDIT: `bind = , mouse:274, exec, ;` should  do the trick.
In programs like kitty, you may also want to change their config to ignore the MMB.

**Reply by raffaelemancuso** [reply]
Hi fufexan,
me do not understand.
I have checked the wiki before posting.
That option is documented as:
> Sending LMB and RMB simultaneously will be interpreted as a middle click. This disables any touchpad area that would normally send a middle click based on location.

I do not send LMB and RMB simultaneously. The middle point in the lower part of my touchpad send MMB directly, apparently.

**Reply by fufexan** [reply]
Ah, I see.

`libinput` doesn't have an option for that, but you could bind that mouse button (find it with `wev`) to do nothing.

**Reply by raffaelemancuso** [reply]
me binding doesn't work:

```
bind = , mouse:274, exec,
```

it still pastes the clipboard.

And where can I find the noop in the wiki? https://wiki.hyprland.org/Configuring/Binds/

**Reply by ghost** [reply]
@fufexan @vaxerski so what is the code to disable middle-mouse button?

**Reply by fufexan** [reply]
`bind=,mouse:274,exec,;` in the config, or
`hyprctl keyword bind ',mouse:274,exec,;'` in your terminal. These work for me.

**Comment by 413x1nkp** [comment]
> `input:touch:middle_button_emulation = 0`.
> 
> It's in the wiki, in Variables, Input section.
> 
> EDIT: `bind = , mouse:274, exec, ;` should do the trick. In programs like kitty, you may also want to change their config to ignore the MMB.

The issue is that this disabled ALL functionality of the middle mouse. Including in the browser, other binds, etc.
I believe the user (and, most of us too) would want a solution that gets rid of middle mouse button paste _while_ preserving the middle mouse button functionality.

**Reply by dantefromhell** [reply]
Yes please.

I had to revert the suggestion because the binding made my middle mouse button unusable in games.

**Reply by Elias-Graf** [reply]
What is the solution to this? I use the middle mouse extensively to open new links in browsers, or to pan in canvases. I only need to disable the paste, not the click itself.

**Reply by Agent00Ming** [reply]
> What is the solution to this? I use the middle mouse extensively to open new links in browsers, or to pan in canvases. I only need to disable the paste, not the click itself.

there's an option in the wiki just for that `misc:middle_click_paste` https://wiki.hyprland.org/Configuring/Variables/#misc

**Reply by fufexan** [reply]
> What is the solution to this? I use the middle mouse extensively to open new links in browsers, or to pan in canvases. I only need to disable the paste, not the click itself.

I also use the middle click for those things, and pasting doesn't get in my way, as it won't try to paste anything when middle clicking a link (at least in Firefox and Chromium).

**Reply by 413x1nkp** [reply]
> I also use the middle click for those things, and pasting doesn't get in my way, as it won't try to paste anything when middle clicking a link (at least in Firefox and Chromium).

i was used to using middle mouse to autoscroll (hold MMB and move cursor to move on the page) when i was on windows, when i switched over i had to resort to using PgUp and PgDw keys, but those are extremely imprecise and just don't feel as good to use...

**Reply by fufexan** [reply]
Autoscroll works on Linux just fine. On Firefox it has to be enabled in settings, and on Chromium you have to launch it with `--enable-blink-features=MiddleClickAutoscroll`.

**Reply by 413x1nkp** [reply]
> Autoscroll works on Linux just fine. On Firefox it has to be enabled in settings, and on Chromium you have to launch it with `--enable-blink-features=MiddleClickAutoscroll`.

OH MY GOD YOU'RE RIGHT

I'm so stupid, I'm sorry :sob: 

The setting is `general.autoScroll` in about:config
